src/Avalonia.Controls/TextBox.cs

```diff
@@ -655,6 +655,8 @@ protected override void OnPointerMoved(PointerEventArgs e)
             if (_presenter != null && e.Device.Captured == _presenter)
             {
                 var point = e.GetPosition(_presenter);
+
+                point = new Point(Math.Max(Math.Min(point.X, _presenter.Bounds.Width - 1), 0), Math.Max(Math.Min(point.Y, _presenter.Bounds.Height - 1), 0));
```

Comment:
Instead of `Math.Max(Math.Min)` here you could use `MathUtilities.Clamp` (also: netcore 2.1 finally adds `Math.Clamp`!)